test/e2e_awssharedvpc/aws_shared_vpc_test.go

```diff
@@ -0,0 +1,291 @@
+		if err != nil {
+			t.Logf("Failed to get DNS record for shared VPC zone: %v", err)
+			return false, nil
+		} else if len(recordValues) == 0 {
+			t.Logf("No DNS records with name %q", expectedHost)
+			return false, nil
+		}
+
+		if _, found := recordValues[serviceAddress]; !found {
+			if _, foundWithDot := recordValues[serviceAddress+"."]; !foundWithDot {
+				t.Logf("DNS record with name %q didn't contain expected service IP %q", expectedHost, serviceAddress)
+				return false, nil
+			}
+		}
+		t.Logf("DNS record with name %q found in shared Route 53 private zone %q and matched service IPs", expectedHost, hostedZoneID)
+		return true, nil
+	}); err != nil {
+		t.Fatal("Failed to verify that DNS has been correctly set in a different account.")
+	}
+
+	t.Log("Querying for DNS record inside cluster VPC using a dig pod")
+
+	// Verify that the IPs of the record created by ExternalDNS match the IPs of load balancer obtained in the previous
+	// step. We will start a pod that runs a dig command for the expected hostname and parse the dig output.
+	if err := wait.PollUntilContextTimeout(context.TODO(), common.DnsPollingInterval, common.DnsPollingTimeout, true, func(ctx context.Context) (done bool, err error) {
+		gotIPs, err := common.LookupARecordInternal(ctx, t, testNamespace, expectedHost)
+		if err != nil {
+			t.Logf("Failed to query hostname inside cluster: %v", err)
+			return false, nil
+		}
+
+		// If all IPs of the loadbalancer are not present, query again.
+		if len(gotIPs) == 0 {
+			t.Log("Failed to resolve any IPs for the DNS record, retrying...")
+			return false, nil
+		}
+		if len(gotIPs) < len(serviceIPs) {
+			t.Logf("Expected %d IPs, but got %d, retrying...", len(serviceIPs), len(gotIPs))
+			return false, nil
+		}
+		// All expected IPs should be in the received IPs,
+		// but these 2 sets are not necessary equal.
+		for ip := range serviceIPs {
+			if _, found := gotIPs[ip]; !found {
+				return false, nil
+			}
+		}
+		t.Log("Expected IPs are equal to IPs resolved.")
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Failed to verify that DNS has been correctly set: %v", err)
+	}
+
+	// Delete the service and make sure ExternalDNS cleans up the DNS Records
+	t.Log("Deleting service and verifying ExternalDNS deletes DNS records.")
+	if err = common.KubeClient.Delete(context.TODO(), service); err != nil {
+		t.Fatalf("Error deleting service %s/%s: %v", service.Namespace, service.Name, err)
+	}
+	if err = wait.PollUntilContextTimeout(context.TODO(), common.DnsPollingInterval, common.DnsPollingTimeout, true, func(ctx context.Context) (done bool, err error) {
+		recordValues, err := getResourceRecordValues(hostedZoneID, expectedHost, "A")
+		if len(recordValues) != 0 {
+			t.Logf("Waiting for DNS record %q to be deleted.", expectedHost)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Failed to verify that ExternalDNS deleted DNS records: %v", err)
+	}
```

Comment:
I think this should be wrapped into `defer` statement right after the service creation. Otherwise we risk to not make sure the DNS disappers if the test `Fatal`s before reaching the end.

Reply:
Ah yea you are right. Done.